Libraries/Components/ScrollResponder.js

```diff
@@ -406,7 +407,11 @@ var ScrollResponderMixin = {
       if (this.preventNegativeScrollOffset) {
         scrollOffsetY = Math.max(0, scrollOffsetY);
       }
-      this.scrollResponderScrollTo(0, scrollOffsetY);
+      //For horizontal scroll enabled scroll views, current x position of the scroll view needs to be taken into account to
+      // preserve the horizontal scroll offset and prevent the scroll view from scrolling back.
+      RCTScrollViewManager.getScrollViewPosition(React.findNodeHandle(this) ,(x, y)=>{
```

Comment:
Can you fix the spacing?  Space should be after the comma, not before, and `=>` should have space around it.